packages/erc3k/contracts/ERC3000.sol

```diff
@@ -5,21 +5,66 @@
 pragma solidity ^0.6.8;
 pragma experimental ABIEncoderV2;
 
-import "@aragon/govern-contract-utils/contracts/erc165/ERC165.sol";
+import "./ERC3000Data.sol";
+import "./ERC3000Executor.sol";
 
-import "./IERC3000.sol";
-
-abstract contract ERC3000 is IERC3000, ERC165 {
+abstract contract ERC3000 {
     bytes4 internal constant ERC3000_INTERFACE_ID =
-        this.schedule.selector
-        ^ this.execute.selector
-        ^ this.challenge.selector
-        ^ this.resolve.selector
-        ^ this.veto.selector
-        ^ this.configure.selector
-    ;
-
-    function supportsInterface(bytes4 _interfaceId) virtual override public pure returns (bool) {
-        return _interfaceId == ERC3000_INTERFACE_ID || super.supportsInterface(_interfaceId);
-    }
+          this.schedule.selector
+          ^ this.execute.selector
+          ^ this.challenge.selector
+          ^ this.resolve.selector
+          ^ this.veto.selector
+          ^ this.configure.selector
+        ;
+
+    /**
+     * @notice Schedules an action for execution, allowing for challenges and vetos on a defined time window
+     * @param container A Container struct holding both the paylaod being scheduled for execution and
+       the current configuration of the system
+     */
+    function schedule(ERC3000Data.Container memory container) virtual public returns (bytes32 containerHash);
+    event Scheduled(bytes32 indexed containerHash, ERC3000Data.Payload payload, ERC3000Data.Collateral collateral);
+
+    /**
+     * @notice Executes an action after its execution delayed has passed and its state hasn't been altered by a challenge or veto
+     * @param container A ERC3000Data.Container struct holding both the paylaod being scheduled for execution and
+       the current configuration of the system
+     * should be a MUST payload.executor.exec(payload.actions)
+     */
+    function execute(ERC3000Data.Container memory container) virtual public returns (bytes32 failureMap, bytes[] memory execResults);
+    event Executed(bytes32 indexed containerHash, address indexed actor);
+
+    /**
+     * @notice Challenge a container in case its scheduling is illegal as per Config.rules. Pulls collateral and dispute fees from sender into contract
+     * @param container A ERC3000Data.Container struct holding both the paylaod being scheduled for execution and
+       the current configuration of the system
+     * @param reason Hint for case reviewers as to why the scheduled container is illegal
```

Comment:
Maybe we could expand this to make it clear that you could, for example, submit an IPFS CID for the dispute resolution system to pick up and use as the challenger's argument.